cmd/libvirt-actuator/resources/init.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+
+# Update the qemu-kvm to a version that has -fw_cfg flag (2.4 at least)
+cat <<EOF |
+[centos]
+name=centos
+baseurl=http://mirror.centos.org/centos/7/virt/x86_64/kvm-common/
+gpgcheck=0
+enabled=1
+EOF
+ssh libvirtactuator 'cat > /etc/yum.repos.d/centos.repo'
+
+ssh libvirtactuator 'yum update qemu-kvm -y'
+
+# Create default storage volume
+export VIRSH="virsh -c qemu+ssh://root@libvirtactuator/system"
+
+${VIRSH} pool-define /dev/stdin <<EOF
+<pool type='dir'>
+  <name>default</name>
+  <target>
+    <path>/var/lib/libvirt/images</path>
+  </target>
+</pool>
+EOF
+
+${VIRSH} pool-start default
+${VIRSH} pool-autostart default
+
+script_dir="$(cd $(dirname "${BASH_SOURCE[0]}") && pwd -P)"
+pushd ${script_dir}
+# Upload cloud-init iso to bootstrap an instance
+scp cloud-init.iso libvirtactuator:/var/lib/libvirt/images/cloud-init.iso
+popd
+
+ssh libvirtactuator 'wget --inet4-only https://download.fedoraproject.org/pub/fedora/linux/releases/28/Cloud/x86_64/images/Fedora-Cloud-Base-28-1.1.x86_64.qcow2'
+# Took from https://askubuntu.com/a/299578
+ssh libvirtactuator 'virsh vol-create-as default fedora_base $(stat -Lc%s Fedora-Cloud-Base-28-1.1.x86_64.qcow2) --format raw'
+ssh libvirtactuator 'virsh vol-upload --pool default fedora_base Fedora-Cloud-Base-28-1.1.x86_64.qcow2'
```

Comment:
Ah, so we use `vol-upload` here. Hence my previous comment about using it vis-a-vis `sudo cp`.